File: BaekJoon/5052.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

bool compareNumbers(vector<string> &phoneNumbers, int numberCount)
{
  for (int i = 0; i < numberCount; i++)
  {
    for (int j = i + 1; j < numberCount; j++)
    {
      bool inconsistent = true;
      for (int k = 0; k < phoneNumbers[i].length(); k++)
      {
        if (phoneNumbers[i][k] != phoneNumbers[j][k])
        {
          inconsistent = false;
          break;
        }
      }
      if (inconsistent)
      {
        return false;
      }
    }
  }
  return true;
}

int main()
{
  // freopen("input.txt", "r", stdin);
  int testCase;
  cin >> testCase;
  while (testCase--)
  {
    int numberCount;
    cin >> numberCount;
    vector<string> phoneNumbers(numberCount);
    for (int i = 0; i < numberCount; i++)
    {
      cin >> phoneNumbers[i];
    }
    sort(phoneNumbers.begin(), phoneNumbers.end());
    cout << (compareNumbers(phoneNumbers, numberCount) ? "YES\n" : "NO\n");
  }
}

File: BaekJoon/5052.py
import sys

# sys.stdin = open("input.txt", "r")


class Node:
    def __init__(self):
        self.children = [None] * 10
        self.end = False


def check(count):
    number_list = [list(map(int, input())) for _ in range(count)]
    phone_numbers = Node()

    for numbers in number_list:
        current = phone_numbers

        for number in numbers:
            if current.children[number] is None:
                current.children[number] = Node()

            current = current.children[number]
            if current.end:
                return False

        if any(current.children):
            return False
        current.end = True

    return True


t = int(input())

for _ in range(t):
    print("YES" if check(int(input())) else "NO")
